examples/react-full/pages/star-wars/index/index.page.server.ts

```diff
@@ -1,11 +1,16 @@
 import fetch from 'node-fetch'
+import { withTypescript } from 'vite-plugin-ssr'
 import { filterMovieData } from '../filterMovieData'
 import type { Movie, MovieDetails } from '../types'
 
 export { onBeforeRender }
 export { prerender }
 
-async function onBeforeRender() {
+const onBeforeRender = withTypescript('onBeforeRender', async (pageContext) => {
```

Comment:
Seems like something is broken; TS complains that the function doesn't match what `withTypescript` expects.

Reply:
There seems to be some conflict between default and extended types. I can fix that by naming Internal and External typing different and merging them into a third type